test/uk/gov/hmrc/merchandiseinbaggageinternalfrontend/support/DeclarationJourneyControllerSpec.scala

```
@@ -0,0 +1,81 @@
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait DeclarationJourneyControllerSpec extends BaseSpecWithApplication {
+  lazy val injector: Injector = app.injector
+  implicit lazy val appConf: AppConfig = injector.instanceOf[AppConfig]
+
+  lazy val component: MessagesControllerComponents = injector.instanceOf[MessagesControllerComponents]
+  lazy val strideAuth: StrideAuthAction = injector.instanceOf[StrideAuthAction]
+  lazy val messageApi: Map[String, String] = injector.instanceOf[MessagesApi].messages("default")
+  lazy val messages: Messages = injector.instanceOf[MessagesApi].preferred(Seq(Lang("en")))
+
+  lazy val repo = injector.instanceOf[DeclarationJourneyRepository]
+  lazy val defaultBuilder = injector.instanceOf[DefaultActionBuilder]
+  lazy val actionProvider = injector.instanceOf[DeclarationJourneyActionProvider]
+  lazy val addressLookupConnector = injector.instanceOf[AddressLookupFrontendConnector]
+  lazy val calculationService = injector.instanceOf[CalculationService]
+  lazy val tpsPaymentsService = injector.instanceOf[TpsPaymentsService]
+  lazy val mibConnector = injector.instanceOf[MibConnector]
+
+  def buildPost(url: String, sessionId: SessionId = SessionId("123")): FakeRequest[AnyContentAsEmpty.type] =
+    buildRequest(POST, url, sessionId)
+
+  def buildGet(url: String, sessionId: SessionId = SessionId("123")): FakeRequest[AnyContentAsEmpty.type] =
+    buildRequest(GET, url, sessionId)
+
+  private def buildRequest(httpVerbs: String, url: String, sessionId: SessionId): FakeRequest[AnyContentAsEmpty.type] =
+    FakeRequest(httpVerbs, url)
+      .withSession(SessionKeys.sessionId -> sessionId.value)
+      .withCSRFToken
+      .asInstanceOf[FakeRequest[AnyContentAsEmpty.type]]
+
+  val stubRepo: DeclarationJourney => DeclarationJourneyRepository = declarationJourney =>
+    new DeclarationJourneyRepository(app.injector.instanceOf[ReactiveMongoComponent].mongoConnector.db) {
```

you can just create this `app.injector.instanceOf[ReactiveMongoComponent].mongoConnector.db` once and use it here , instead of executing it each time ?

---

instead of stub, if we use mock then the test case fails if the db method is not called which is a good practice, but that's fine as you have already followed this approach

---

Yes on 1st comment, big no for me on mock Vs stub. Mock needs 1. library 2. gives very weird error when fails

---

BTW will fail without mock as well....